x-pack/test/alerting_api_integration/spaces_only/tests/alerting/group1/get_rule_tags.ts

```diff
@@ -142,60 +142,116 @@ export default function createAggregateTests({ getService }: FtrProviderContext)
+          per_page: 5,
+          page: 1,
+          total: 16,
+        });
 
-      response = await supertest
-        .get(
-          `${getUrlPrefix(
-            Spaces.space1.id
-          )}/internal/alerting/rules/_tags?page=2&per_page=5&search=1`
-        )
-        .expect(200);
+        response = await supertest
+          .get(
+            `${getUrlPrefix(
+              Spaces.space1.id
+            )}/internal/alerting/rules/_tags?page=2&per_page=5&search=1`
+          )
+          .expect(200);
+
+        expect(response.body).to.eql({
+          data: ['1111', '11111', '11_11', '12', '13'],
+          per_page: 5,
+          page: 2,
+          total: 16,
+        });
 
-      expect(response.body).to.eql({
-        data: ['1111', '11111', '11_11', '12', '13'],
-        per_page: 5,
-        page: 2,
-        total: 16,
+        response = await supertest
+          .get(
+            `${getUrlPrefix(
+              Spaces.space1.id
+            )}/internal/alerting/rules/_tags?page=1&per_page=5&search=11`
+          )
+          .expect(200);
+
+        expect(response.body).to.eql({
+          data: ['11', '110', '111', '1111', '11111'],
+          per_page: 5,
+          page: 1,
+          total: 6,
+        });
       });
 
-      response = await supertest
-        .get(
-          `${getUrlPrefix(
-            Spaces.space1.id
-          )}/internal/alerting/rules/_tags?page=1&per_page=5&search=11`
-        )
-        .expect(200);
+      describe('by rule type ids', () => {
+        it('should return no tags when no rule type id matches', async () => {
+          await createRule({
+            tags,
+          });
+
+          const response = await supertest
+            .get(
+              `${getUrlPrefix(
+                Spaces.space1.id
+              )}/internal/alerting/rules/_tags?page=1&per_page=5&rule_type_ids=["non_existent"]`
+            )
+            .expect(200);
+
+          expect(response.body).to.eql({
+            data: [],
+            per_page: 5,
+            page: 1,
+            total: 0,
+          });
+        });
 
-      expect(response.body).to.eql({
-        data: ['11', '110', '111', '1111', '11111'],
-        per_page: 5,
-        page: 1,
-        total: 6,
+        it('should return only tags matching rule type ids', async () => {
+          await createRule({
+            tags: ['rule1-tag1', 'rule1-tag2'],
+          });
+
+          await createRule({
+            tags: ['rule2-tag1', 'rule2-tag2'],
+            rule_type_id: 'test.restricted-noop',
+          });
+
+          await createRule({
+            tags: ['rule3-tag1', 'rule3-tag2'],
+            rule_type_id: 'test.unrestricted-noop',
+          });
+
+          const response = await supertest
+            .get(
+              `${getUrlPrefix(
+                Spaces.space1.id
+              )}/internal/alerting/rules/_tags?page=1&per_page=5&rule_type_ids=["test.noop","test.restricted-noop"]`
```

Comment:
nit: Let's increase the `per_page` to a number bigger than the total tags to be sure they are not returned due to filtering and not due to pagination limitations.